pkg/queryfrontend/config.go

```diff
@@ -256,7 +256,9 @@ func (cfg *Config) Validate() error {
 		if cfg.QueryRangeConfig.SplitQueriesByInterval <= 0 && !cfg.isDynamicSplitSet() {
 			return errors.New("split queries or split threshold interval should be greater than 0 when caching is enabled")
 		}
-		if err := cfg.QueryRangeConfig.ResultsCacheConfig.Validate(querier.Config{}); err != nil {
+		if err := cfg.QueryRangeConfig.ResultsCacheConfig.Validate(querier.Config{
+			EnablePerStepStats: cfg.QueryRangeConfig.ResultsCacheConfig.CacheQueryableSamplesStats,
+		}); err != nil {
```

Comment:
Had to do this because of validation of parameters in Cortex QFE code. They compare QFE config against Cortex Querier config. But we ignore that, so I had to force that to pass validation. Or I can delete the validation piece, but then we will have to touch vendored code.

Reply:
Maybe we can extract the config in a variable to make it a bit more readable, and add a short comment above it.

Reply:
Good idea. done